app/src/main/java/com/nevilleantony/prototype/downloadmanager/DownloadRepo.java

```diff
@@ -0,0 +1,119 @@
+package com.nevilleantony.prototype.downloadmanager;
+
+import android.content.Context;
+import android.os.Environment;
+import android.util.Log;
+
+import com.nevilleantony.prototype.storage.DownloadsModel;
+import com.nevilleantony.prototype.storage.StorageApi;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class DownloadRepo {
+    private static final String PATH = Environment.getExternalStorageDirectory().getAbsolutePath() + "/Prototype/";
+    private static DownloadRepo downloadRepo = null;
+    private Map<String, FileDownload> downloadMap = new HashMap<>();
+    private StorageApi stgApi = new StorageApi();
+
+    private DownloadRepo(Context context) {
+        File path = new File(PATH);
+        if (!path.exists()) {
+            path.mkdir();
+        }
+        stgApi.setContext(context);
+        List<DownloadsModel> downloadsModelList = stgApi.getAll();
```

Comment:
Here and everywhere else where this is called, if context is needed pass it as an argument to each method instead of having another method for just setting a context. And as said before this method shouldn't be required.